test/core/pipeline/test_pipeline.py

```diff
@@ -1008,3 +1008,13 @@ def fake_generator_run(self, prompt: str, generation_kwargs: Optional[Dict[str,
     res = pipe.run({"prompt_builder": {"query": "What is the capital of Italy?"}})
 
     assert res == {"router": {"correct_replies": ["Rome"]}}
+
+
+def test__init_graph():
+    pipe = Pipeline()
+    pipe.add_component("greet", Greet())
+    pipe.add_component("adder", AddFixedValue())
+    pipe.connect("greet", "adder")
+    pipe._init_graph()
+    for node in pipe.graph.nodes:
+        assert pipe.graph.nodes[node]["visits"] == 0
```

Comment:
Should I do the same for the newly added private methods?

Reply:
Yeah, that would be cool.